### DISCUSSION TITLE
Control a FIFO stream externally

### DISCUSSION THREAD

**Discussion by capocasa** [streamlink/streamlink#5166] [Q&A] [answered]
### Description

I am successfully streaming content to the snapcast local network streamer, yay! I set up snapcast to accept raw PCM data through a named pipe at `/tmp/snapfifo` and the following call works great:

`streamlink -o - mysite.com/mystream audio | ffmpeg -i - -f s16le -acodec pcm_s16le pipe:1 > /tmp/snapfifo
`
Normally I use snapcast with MPD and spotify, both of which allow having a stream much like the one above, but controlling it using some kind of GUI- in MPD's one of any number of control interfaces that communicate with MPD through a standardized API, and in Spotify's case the feature built into the proprietary app.

Is there some way to control an ongoing stream, such as skipping (where appropriate) or even changing the source, when the stream is already running?

My next steps might be to cobble something together using a script that stops the stream and starts it again, either in Python or Shell. But if something like this already exists, either as core functionality or as part of a plugin or 3rd party app, I'd much rather leverage that instead.

**Comment by bastimeyer** [comment, marked as answer]
> Is there some way to control an ongoing stream, such as skipping (where appropriate) or even changing the source, when the stream is already running?

Streamlink doesn't (and is unable to) support any of that. Streams can't be skipped or switched. Once you start Streamlink, its output will be a continuous stream until it ends.

**Reply by capocasa** [reply]
Thanks for clarifying!

Then it's down to possibly discovering kind of third party frontend- or elbow grease.
